src/d3a/models/area/__init__.py

```diff
@@ -393,13 +391,17 @@ def tick(self):
             if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
                 self.dispatcher.publish_market_clearing()
             else:
-                for market in self.all_markets:
-                    bid_offer_pairs = self.bid_offer_matcher.calculate_recommendation(
-                        *market.open_bids_and_offers, self.now)
-                    while bid_offer_pairs:
-                        bid_offer_pairs = self.bid_offer_matcher.calculate_recommendation(
+                if is_external_matching_enabled():
+                    bid_offer_matcher.match_algorithm.area_uuid_markets_mapping.\
+                        update({self.uuid: self.all_markets})
```

Comment:
Could you please explain what happens here?
I'm not very convinced about this part: it interacts a bit too deeply with the internals of the matcher (without using any interface method). This makes testing trickier, among other stuff. I think it would be cool if we could refactor this somehow, but not necessarily in the context of this PR. Maybe for the next test, if that's ok with you.

Reply:
This is updating the mapping of areas: open offers / bids
I agree with you that this should be using an interface, I'm touching this part in my current test to only update if the Area is not a device, I'll also add the interface there too
Thanks